### PR TITLE
Updates to grunt typedoc

### PR DESCRIPTION
**Pull request by devpaul**
**Type:** feature

The following has been addressed in the PR:

* support publishing of typedocs w/ an encrypted deploy key
* Added documentation to README.md
* cleanup and tests

**Description:**

Resolves #55849 & #55847

### CODE REVIEW

File: tasks/typedoc.ts [hunk 1]
```diff
@@ -1,97 +1,52 @@
...
+	}).forEach(key => {
+		if (typeof options[key] === 'boolean') {
+			if (options[key]) {
+				args.push('--' + key);
+			}
+		}
+		else if (options[key]) {
+			args.push('--' + key);
+			args.push(`"${options[key]}"`);
+		}
```
Comment (devpaul) [inline tasks/typedoc.ts]: Good call. I'll update!

**Comment by devpaul** [comment]
Verified integration works. See https://travis-ci.org/devpaul/core and http://devpaul.github.io/core/api/

**Comment by agubler** [comment]
@devpaul I get 404's for all the links on in the api docs you published.... is that expected?

**Comment by devpaul** [comment]
*sigh* kinda. I need to add a no Jekyll file. https://help.github.com/articles/files-that-start-with-an-underscore-are-missing/

**Comment by agubler** [comment]
@devpaul is there anything more you need to do on this one?

**Comment by devpaul** [comment]
To get this working on GitHub pages I need to add a nojekyll file so it
publishes pages beginning with an underscore

**Pull request merged**